client/network/src/protocol/notifications/upgrade/notifications.rs

```diff
@@ -106,51 +108,58 @@ pub struct NotificationsOutSubstream<TSubstream> {
 
 impl NotificationsIn {
 	/// Builds a new potential upgrade.
-	pub fn new(protocol_name: impl Into<Cow<'static, str>>, max_notification_size: u64) -> Self {
+	pub fn new(
+		main_protocol_name: impl Into<Cow<'static, str>>,
+		fallback_names: Vec<Cow<'static, str>>,
+		max_notification_size: u64
+	) -> Self {
+		let mut protocol_names = fallback_names;
+		if protocol_names.is_empty() {
+			protocol_names.push(main_protocol_name.into())
+		} else {
+			protocol_names.insert(0, main_protocol_name.into());
+		}
```

Comment:
```suggestion
		protocol_names.insert(0, main_protocol_name.into());
```

To reduce cyclomatic complexity

Reply:
I wrote this because I thought `insert(0, ...)` panicked if the `Vec` is empty.

Reply:
Indeed, [it doesn't](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=ddec6a16bf29f8344dca505b3a25c6be).